spec/unit/rendezvous/rendezvous.spec.ts

```diff
@@ -23,75 +23,142 @@ import {
     MSC3903ECDHPayload,
     MSC3903ECDHv2RendezvousChannel as MSC3903ECDHRendezvousChannel,
 } from "../../../src/rendezvous/channels";
-import { MatrixClient } from "../../../src";
+import { Device, MatrixClient } from "../../../src";
 import {
     MSC3886SimpleHttpRendezvousTransport,
     MSC3886SimpleHttpRendezvousTransportDetails,
 } from "../../../src/rendezvous/transports";
 import { DummyTransport } from "./DummyTransport";
 import { decodeBase64 } from "../../../src/base64";
 import { logger } from "../../../src/logger";
-import { DeviceInfo } from "../../../src/crypto/deviceinfo";
+import { CrossSigningKey, OwnDeviceKeys } from "../../../src/crypto-api";
+
+type UserID = string;
+type DeviceID = string;
+type Fingerprint = string;
+type PartialUserDevices = Map<DeviceID, Partial<Device>>;
+type PartialDeviceMap = Map<UserID, PartialUserDevices>;
+type SimpleDeviceMap = Record<UserID, Record<DeviceID, Fingerprint>>;
+
+function mockDevice(userId: UserID, deviceId: DeviceID, fingerprint: Fingerprint): Device {
+    return {
+        deviceId,
+        userId,
+        getFingerprint: () => fingerprint,
+    } as unknown as Device;
+}
+
+function mockDeviceMap(
+    userId: UserID,
+    deviceId: DeviceID,
+    deviceKey?: Fingerprint,
+    otherDevices: SimpleDeviceMap = {},
+): PartialDeviceMap {
+    const deviceMap: PartialDeviceMap = new Map();
+
+    const myDevices: PartialUserDevices = new Map();
```

Comment:
Now that `mockDevice` returns a `Device`, `myDevices` should now be a regular `Map<string, Device>` rather than a `PartialUserDevices`. 

Likewise, `deviceMap` should be a `Map<string, Map<string, Device>>`.

And we can update the return type of this and `makeMockClient`.

And then `PartialUserDevices` and `PartialDeviceMap` are unused.